homeassistant/components/broadlink/datetime.py

```diff
@@ -0,0 +1,77 @@
+"""Support for Broadlink sensors."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from homeassistant.components.datetime import DateTimeEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .entity import BroadlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Broadlink datetime."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+    async_add_entities([BroadlinkDateTime(device)])
+
+
+class BroadlinkDateTime(BroadlinkEntity, DateTimeEntity):
+    """Representation of a Broadlink date and time."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, device) -> None:
+        """Initialize the sensor."""
+        super().__init__(device)
+
+        self._attr_unique_id = f"{device.unique_id}-datetime"
+        self._update_internal_state()
```

Comment:
We don't need to update the internal state here, the coordinator will push the first state to the entity.